types/src/v0/impls/stake_table.rs

```diff
@@ -777,63 +781,72 @@ impl Fetcher {
             Some(chunk)
         });
 
+        let max_retry_duration = l1_client.options().l1_events_max_retry_duration;
         // fetch registered events
         // retry if the call to the provider to fetch the events fails
         let registered_events = stream::iter(chunks.clone()).then(|(from, to)| {
             let retry_delay = l1_client.options().l1_retry_delay;
             let stake_table_contract = stake_table_contract.clone();
             async move {
                 tracing::debug!(from, to, "fetch ValidatorRegistered events in range");
-                loop {
-                    match stake_table_contract
-                        .clone()
-                        .ValidatorRegistered_filter()
-                        .from_block(from)
-                        .to_block(to)
-                        .query()
-                        .await
-                    {
-                        Ok(events) => break stream::iter(events),
-                        Err(err) => {
-                            tracing::warn!(from, to, %err, "ValidatorRegistered Error");
-                            sleep(retry_delay).await;
-                        },
-                    }
-                }
+                let events = retry(
```

Comment:
We still have an infinite loop on the `initializedAtBlock` call. Should we panic there too? It doesn't involve the L1 events range but if the RPC is misconfigured or broken it would now hang forever.